tools/bin/ecflow/processes/main/meteorological/chunk_forcings.py

```diff
@@ -0,0 +1,48 @@
+''' chunk_forcings.py
+usage: python save_metsim_by_year.py <metsim_config_file>
+Reads in output from MetSim (corresponding to information in
+metsim_config_file) and writes it to separate files by year for use in VIC '''
+import os
+import sys
+import argparse
+import numpy as np
+import xarray as xr
+
+from tonic.io import read_config
+
+
+def save_metsim_by_year(metsim_config_file):
+    ''' Reads MetSim output and writes it to separate files by calendar
+        year so that they can be read to force VIC '''
+    # read configuration file to get file name information
+    config_dict = read_config(metsim_config_file)
+    out_dir = config_dict['MetSim']['out_dir']
+    out_prefix = config_dict['MetSim']['out_prefix']
+    start = config_dict['MetSim']['start'].replace('/','')
+    stop = config_dict['MetSim']['stop'].replace('/','')
+    in_file = os.path.join(out_dir, '{0}_{1}-{2}.nc'.format(out_prefix,
+                                                          start, stop))
+    # read data and save by year
+    if os.path.exists(in_file):
+        xds =xr.open_dataset(in_file)
+        for year in np.unique(xds.time.dt.year):
+            out_file = os.path.join(out_dir, '{0}_{1}-{2}.{3}.nc'.format(
+                out_prefix, start, stop, year))
+            xds.sel(time=str(year)).to_netcdf(out_file)
```

Comment:
I wonder if the xarray.groupy('time.year') could achieve this

Reply:
My way was very slightly faster in testing.